bindings/experimental/distrdf/python/DistRDF/Backends/Utils.py

```diff
@@ -241,3 +241,34 @@ def _(mergeable: ROOT.Detail.RDF.RMergeableVariationsBase, node, backend):
     the RMergeableVariations class.
     """
     node.value = mergeable
+
+
+@singledispatch
+def clone_action(resultptr):
+    """
+    Clone the action held by an RResultPtr, registering it with its RLoopManager
+    """
+    return ROOT.Internal.RDF.CloneResultPtr(resultptr)
+
+
+@clone_action.register(AsNumpyResult)
+def _(asnumpyres):
+    asnumpyres._result_ptrs = {col: ROOT.Internal.RDF.CloneResultPtr(ptr) for (col, ptr) in asnumpyres._result_ptrs.items()}
+    return asnumpyres
+
+
+def clone_snapshotresult(snap: SnapshotResult, range_id: int):
+    """
+    Clone the action held by an RResultPtr, registering it with its RLoopManager
+    """
+    name_with_old_id = snap.filenames[0].removesuffix(".root")
```

Comment:
what happens if the output file name does not end in `.root`? If it's required that it ends in `.root`, are we checking for that anywhere?

Reply:
Indeed it should not be an issue, I extended the logic to also accomodate files that do not end in `.root`